File: Searching/Binary_Search.py
def binarySearch(element, l, r, arr):
    if r >= l:
        mid = l + (r-l)//2
        if arr[mid] == element:
            return f"Element found at {mid} position"
        elif arr[mid] > element:
            return binarySearch(element, l, mid-1, arr)
        else:
            return binarySearch(element, mid+1, r, arr)
    else:
        return f"Element not found"

if __name__ == "__main__":
    arr = [-45, -1, 9, 9, 11, 46, 89, 1000]
    # for ele in arr:
    #     print(binarySearch(ele, 0, len(arr), arr))
    print(binarySearch(-46, 0, len(arr), arr))

File: Javascript/binary_search.js
// iterative function to implement Binary Search
const binary_search = (arr, x) => {
  let start = 0,
    end = arr.length - 1;

  // iterate while start not meets end
  while (start <= end) {
    // find the mid index
    let mid = Math.floor((start + end) / 2);

    // if element is present at mid, return true
    if (arr[mid] === x) {
      return true;
    } else if (arr[mid] < x) {
      start = mid + 1;
    } else {
      end = mid - 1;
    }
  }

  return false;
};

// driver code
const arr = [1, 3, 5, 7, 8, 9];
const x = 5;

if (binary_search(arr, x)) {
  console.log('Element found!');
} else {
  console.log('Element not found!');
}
